components/calendar/src/calendar_arithmetic.rs

```diff
@@ -154,6 +163,7 @@ pub(crate) trait DateFieldsResolver: Calendar {
 }
 
 impl<C: DateFieldsResolver> ArithmeticDate<C> {
+    // Precondition: the date is in the VALID_RD_RANGE
```

Comment:
Nit: This code is complex enough that I'm not convinced that the precondition always hold. Please add a debug-assert. (You can do this in https://github.com/unicode-org/icu4x/pull/7220#discussion_r2496131629)

Reply:
This is moot if my comments on #7220 are resolved.